SpringProject/src/it/java/com/gradproject2019/ConferenceControllerIT.java

```diff
@@ -0,0 +1,173 @@
+package com.gradproject2019;
+
+import com.gradproject2019.conferences.persistance.Conference;
+import com.gradproject2019.conferences.repository.ConferenceRepository;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.time.Instant;
+import java.util.List;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(webEnvironment= SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ConferenceControllerIT {
+
+    @Autowired
+    private ConferenceRepository conferenceRepository;
+
+    @Autowired
+    private TestRestTemplate restTemplate;
+
+    @LocalServerPort
+    int testServerPort;
+
+    @Before
+    public void setUp() {
+       conferenceRepository.deleteAll();
+    }
+
+    @After
+    public void tearDown() {
+       conferenceRepository.deleteAll();
+    }
+
+    @Test
+    public void shouldReturn200AndEmptyListWhenDatabaseEmpty() throws URISyntaxException {
+        //given
+        final String baseUrl = "http://localhost:"+ testServerPort +"/conferences";
+        URI uri = new URI(baseUrl);
+
+        //when
+        ResponseEntity<List<Conference>> response = this.restTemplate.exchange(uri, HttpMethod.GET, null, new ParameterizedTypeReference<List<Conference>>() {});
+
+        //Then
+        Assert.assertEquals(200,response.getStatusCodeValue());
+        Assert.assertEquals(true,response.getBody().isEmpty());
+    }
+
+    @Test
+    public void shouldReturn200AndListOfConferencesWhenDatabasePopulated() throws URISyntaxException {
+        //given
+        final String baseUrl = "http://localhost:"+ testServerPort +"/conferences";
+        URI uri = new URI(baseUrl);
+        Conference conference = new Conference(1L, "Grace's conference", Instant.now(), "Leicester", "All about Grace's fabulous and extra house", "grace");
```

Comment:
Might be best to extract this Conference as a field, as you're using it multiple times